File: HackerEarth/Day 2/Solution.py
def find_max_border(table):
    n, m = len(table), len(table[0])
    max_border = 0

    # Check rows and columns for maximum consecutive black cells
    for row in table:
        max_border = max(max_border, max_consecutive_black(row))
    for col in zip(*table):
        max_border = max(max_border, max_consecutive_black(col))

    return max_border

def max_consecutive_black(cells):
    count = 0
    max_count = 0
    for cell in cells:
        if cell == '#':
            count += 1
            max_count = max(max_count, count)
        else:
            count = 0
    return max_count

# Get test cases and process each one
t = int(input())
for _ in range(t):
    n, m = map(int, input().split())
    table = [input() for _ in range(n)]
    max_border = find_max_border(table)
    print(max_border)

File: HackerEarth/Day 2/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int find_max_border(const vector<string>& table) {
   int n = table.size();
   int m = table[0].size();
   int max_border = 0;

   // Check rows and columns for maximum consecutive black cells
   for (int i = 0; i < n; ++i) {
       max_border = max(max_border, max_consecutive_black(table[i]));
   }

   for (int j = 0; j < m; ++j) {
       string col;
       for (int i = 0; i < n; ++i) {
           col += table[i][j];
       }
       max_border = max(max_border, max_consecutive_black(col));
   }

   return max_border;
}

int max_consecutive_black(const string& cells) {
   int count = 0;
   int max_count = 0;
   for (char cell : cells) {
       if (cell == '#') {
           count++;
           max_count = max(max_count, count);
       } else {
           count = 0;
       }
   }
   return max_count;
}

int main() {
   int t;
   cin >> t;

   for (int _ = 0; _ < t; ++_) {
       int n, m;
       cin >> n >> m;

       vector<string> table(n);
       for (int i = 0; i < n; ++i) {
           cin >> table[i];
       }

       int max_border = find_max_border(table);
       cout << max_border << endl;
   }

   return 0;
}
